### PR TITLE
[NativeAOT-LLVM] publish scenario for nativelib

### PR DESCRIPTION
**Pull request by yowl**
This PR enables the publish scenario for a native lib in WebAssembly.

- Always use an .html extension to make testing libraries easier
- Remove linker options on emcc compile lines - this just removes some emscripten warnings
- Add generation of __managed__Startup when a nativelib
- Include clrjit_browser_wasm32_x64.dll in package

Once these are in you can create a library function, e.g.
```cs
namespace WasmLib;
public class Class1
{
    [System.Runtime.InteropServices.UnmanagedCallersOnly(EntryPoint = "Answer")]
    public static int Answer()
    {
        return 41;
    }
}
```
And publish it with 
```
dotnet publish /p:NativeLib=Static /p:SelfContained=true -r browser-wasm -c Debug /p:TargetArchitecture=wasm /p:PlatformTarget=AnyCPU /p:MSBuildEnableWorkloadResolver=false /p:EmccExtraArgs="-s EXPORTED_FUNCTIONS=_Answer -s EXPORTED_RUNTIME_METHODS=cwrap" --self-contained
```
Then with a bit of added Javascript to the end of the emscripten html file
```js
    <script>
      Module.onRuntimeInitialized = _ => {
        const answer = Module.cwrap('Answer', 'number', []);
        console.log(answer());
      };
    </script>
```
It should be called 
image

cc @SingleAccretion

### CODE REVIEW

File: docs/using-nativeaot/compiling.md [hunk 1]
```diff
@@ -58,17 +58,29 @@ Note that it is important to use _the same version_ for both packages to avoid p
...
-> dotnet publish -r browser-wasm -c Debug /p:TargetArchitecture=wasm
+> dotnet publish -r browser-wasm -c Debug /p:TargetArchitecture=wasm /p:PlatformTarget=AnyCPU
+```
+To compile a WebAssembly native library that exports a function `Answer` :
+```cs
+    [System.Runtime.InteropServices.UnmanagedCallersOnly(EntryPoint = "Answer")]
+    public static int Answer()
+    {
+        return 42;
+    }
```
Comment (SingleAccretion) [inline docs/using-nativeaot/compiling.md]:
```suggestion
[System.Runtime.InteropServices.UnmanagedCallersOnly(EntryPoint = "Answer")]
public static int Answer()
{
    return 42;
}
```

File: docs/using-nativeaot/compiling.md [hunk 2]
```diff
@@ -58,17 +58,29 @@ Note that it is important to use _the same version_ for both packages to avoid p
...
 ```xml
-<PackageReference Include="Microsoft.DotNet.ILCompiler.LLVM; runtime.browser-wasm.Microsoft.DotNet.ILCompiler.LLVM" Version="7.0.0-*" />
+<PackageReference Include="Microsoft.DotNet.ILCompiler.LLVM; runtime.win-x64.Microsoft.DotNet.ILCompiler.LLVM" Version="7.0.0-*" />
 ```
 and the publish command (there is no Release build currently)
 ```bash
-> dotnet publish -r browser-wasm -c Debug /p:TargetArchitecture=wasm
+> dotnet publish -r browser-wasm -c Debug /p:TargetArchitecture=wasm /p:PlatformTarget=AnyCPU
+```
+To compile a WebAssembly native library that exports a function `Answer` :
```
Comment (SingleAccretion) [inline docs/using-nativeaot/compiling.md]:
```suggestion
To compile a WebAssembly native library that exports a function `Answer`:
```

File: docs/using-nativeaot/compiling.md [hunk 3]
```diff
@@ -58,17 +58,29 @@ Note that it is important to use _the same version_ for both packages to avoid p
...
+```cs
+    [System.Runtime.InteropServices.UnmanagedCallersOnly(EntryPoint = "Answer")]
+    public static int Answer()
+    {
+        return 42;
+    }
+```
+```bash
+>dotnet publish /p:NativeLib=Static /p:SelfContained=true -r browser-wasm -c Debug /p:TargetArchitecture=wasm /p:PlatformTarget=AnyCPU /p:MSBuildEnableWorkloadResolver=false /p:EmccExtraArgs="-s EXPORTED_FUNCTIONS=_Answer -s EXPORTED_RUNTIME_METHODS=cwrap" --self-contained
```
Comment (SingleAccretion) [inline docs/using-nativeaot/compiling.md]:
```suggestion
> dotnet publish /p:NativeLib=Static /p:SelfContained=true -r browser-wasm -c Debug /p:TargetArchitecture=wasm /p:PlatformTarget=AnyCPU /p:MSBuildEnableWorkloadResolver=false /p:EmccExtraArgs="-s EXPORTED_FUNCTIONS=_Answer -s EXPORTED_RUNTIME_METHODS=cwrap" --self-contained
```

**Review by SingleAccretion — APPROVED** [review]

**Comment by yowl** [comment]
Thanks

**Comment by yowl** [comment]
@jkotas I think this is ok to be merged when you have a minute, thanks!

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**